app/src/main/java/org/schabi/newpipe/local/history/HistoryRecordManager.java

```diff
@@ -120,19 +120,11 @@ public HistoryRecordManager(final Context context) {
             }
 
             // Update the stream progress to the full duration of the video
-            final List<StreamStateEntity> states = streamStateTable.getState(streamId)
-                    .blockingFirst();
-            if (!states.isEmpty()) {
-                final StreamStateEntity entity = states.get(0);
-                entity.setProgressMillis(duration * 1000);
-                streamStateTable.update(entity);
-            } else {
-                final StreamStateEntity entity = new StreamStateEntity(
-                        streamId,
-                        duration * 1000
-                );
-                streamStateTable.insert(entity);
-            }
+            final StreamStateEntity entity = new StreamStateEntity(
+                    streamId,
+                    duration * 1000
+            );
+            streamStateTable.upsert(entity);
```

Comment:
Nice catch ;-)

Reply:
Yeah, I didn't think to look for an `upsert` method the first time. XD